serverless/functions/serverless-functions-on-cluster-builds.adoc

```diff
@@ -0,0 +1,92 @@
+:_content-type: PROCEDURE
+[id="serverless-functions-on-cluster-builds"]
+= Building and deploying on-cluster functions
+:context: serverless-functions-on-cluster-builds
+include::_attributes/common-attributes.adoc[]
+
+toc::[]
+
+You can use the Knative (`kn`) CLI to push a function project directly to the cluster, which triggers a build and deployment of the function. You can also push a function project to a source code repository branch, which triggers a build of the function with the updated changes and deploys the updated function on the cluster.
+
+:FeatureName: {FunctionsProductName}
+include::snippets/technology-preview.adoc[leveloffset=+1]
+
+[id="prerequisites_serverless-functions-on-cluster-builds"]
+== Prerequisites
+
+* xref:../../cicd/pipelines/installing-pipelines.adoc[{pipelines-title}] must be installed on your cluster.
+
+* You have installed the OpenShift (`oc`) CLI.
+
+* You have installed the Knative (`kn`) CLI.
+
+[id="serverless-functions-on-cluster-builds-procedure"]
+== Procedure
+
+. In each namespace where you want to run {pipelines-shortname} and deploy a function, you must create the following resources:
+
+.. Create the functions buildpacks Tekton task to be able to build the function image:
++
```

Comment:
I believe this is the one I had asked @lance about and he said that it's installed by Pipelines by default. We can't have links to upstream git repos, so if this needs to be included it would need to be available in the `openshift-knative` repo instead of the `tekton` one. @lance PTAL.

Reply:
@abrennan89 is correct that this Task should be already installed with OpenShift Pipelines. This should be confirmed in QA @jrangelramos.

Reply:
Hey @lance .. yea I confirmed that. It looks like that task is not installed by default by Openshift Pipelines. It fails with me on OCP, so I had to install it to get it working.

Reply:
@jrangelramos OK good to know. Perhaps it's part of the catalog. Checking... 

Yes - https://github.com/tektoncd/catalog/tree/main/task/git-clone

This appears to manifest in the OpenShift Developer's Console when creating a pipeline. When you click the "Add task" button, you can choose the `git-clone` task. Here's a screen capture. On this cluster, it appears that the `git-clone` task from Red Hat is already installed.

![image](https://user-images.githubusercontent.com/15952/170284039-335da3f2-8326-4d7e-83ff-3b5ba7d01aa4.png)

There is community version that is not installed here.

![image](https://user-images.githubusercontent.com/15952/170283254-e235b515-2bd8-4ea6-a34d-cb54eb94a8cf.png)

You can install the community version by creating a pipeline which uses it, or directly from the catalog here https://hub.tekton.dev/tekton/Task/git-clone/0.6.

I didn't create the cluster, so it's possible that someone else manually installed the Red Hat `Task` manually by creating a pipeline. Or there may be some way to install a `Task` from the catalog without having to create a pipeline. I'm not sure.

I don't know the best way to document this. Is there any precedent here @abrennan89? Have we documented installing and specific `Task` dependencies yet?

Reply:
No we haven't. I believe the correct thing to do would be to make sure it's documented in the Pipelines docs https://docs.openshift.com/container-platform/4.10/cicd/pipelines/working-with-pipelines-using-the-developer-perspective.html but it's probably a question for @Preeticp

Reply:
Hi @piyush-garg, @nikhil-thomas it would be great if you could provide some clarity wrt to the following questions: 

1. Is there a downstream equivalent for this: https://raw.githubusercontent.com/tektoncd/catalog/master/task/git-clone/0.4/git-clone.yaml? 
2. Is the tekton task git-clone installed by default when you install the Pipelines Operator?

Reply:
WRT:

> No we haven't. I believe the correct thing to do would be to make sure it's documented in the Pipelines docs https://docs.openshift.com/container-platform/4.10/cicd/pipelines/working-with-pipelines-using-the-developer-perspective.html but it's probably a question for @Preeticp

I believe it is documented in step 2 B here: https://docs.openshift.com/container-platform/4.10/cicd/pipelines/working-with-pipelines-using-the-developer-perspective.html#op-constructing-pipelines-using-pipeline-builder_working-with-pipelines-using-the-developer-perspective

Reply:
1. We don't have any downstream equivalent for the git-clone task, we ship the same upstream task in the operator and fetched from upstream catalog https://github.com/tektoncd/operator/blob/main/hack/openshift/fetch-tektoncd-catalog-tasks.sh#L47 However the git-init image which gets used is from redhat, but it uses the same binary from upstream for git-init and just based on Ubi.

2. This task gets installed with openshift pipelines operator given the clustertasks are enabled in tekton config https://github.com/tektoncd/operator/blob/main/docs/TektonConfig.md#addon

Reply:
One more thing is this task get installed by operator as clustertask so it will be available in all namespaces

Reply:
@jrangelramos would it be possible for you to share the Pipeline with @piyush-garg, please? That would help him dig a bit deeper to resolve the issue. Thank you!

Reply:
Additional question - does anyone know what OCP version added this task for the Operator? Our docs for serverless are supported on 4.6+, but I wonder if that's also the case for this Tekton part.

Reply:
@jrangelramos thanks for that update - I understand now. Where do we stand with docs?